### PR TITLE
feat: Add CADDY_EXTRA_DIRECTIVES

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Review by dunglas — CHANGES_REQUESTED** [review]
Good idea. Maybe just `EXTRA_DIRECTIVES` for consistency with `GLOBAL_OPTIONS` which doesn't include the CADDY prefix?

File: charts/mercure/templates/deployment.yaml [hunk 1]
```diff
@@ -47,6 +47,11 @@ spec:
                 configMapKeyRef:
                   name: {{ include "mercure.fullname" . }}
                   key: global-options
+            - name: EXTRA_DIRECTIVES
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ include "mercure.fullname" . }}
+                  key: extra-directives
```
Comment (dunglas) [inline charts/mercure/templates/deployment.yaml:54]: The config template must be updated as well isn't it? We also need a new value.

Also, I would use a secret as this env var may contains secret keys and things like this, depending on the modules you use.

WDYT?

Comment (blackheaven) [reply charts/mercure/templates/deployment.yaml]: I did it, but it may be confusing with `extra-directives`/`extraDirectives`, should I change it too?

**Review by dunglas — APPROVED** [review]

**Comment by blackheaven** [comment]
Actually, it works on my computer, can you have a look please?

**Comment by dunglas** [comment]
I'm not sure of waht the issue is, but the error in the CI is "wget: bad address 'mercure-jg4qdqb8ei:80'".

**Comment by dunglas** [comment]
Thanks!

**Pull request merged**